app/services/weets.js

```diff
@@ -1,10 +1,24 @@
 const api = require('../../config/axios');
+const db = require('../models');
+const errors = require('../errors');
 
-const fetchWeet = () =>
-  api
-    .get('/random')
-    .then(({ data }) => data)
-    .catch(error => Promise.reject(error));
+const fetchWeet = async user => {
+  try {
+    const { data } = await api.get('/random');
+    if (data.length > 140) {
+      return Promise.reject(
+        errors.defaultError('The content of the weet exceeds 140 characters. Try again.')
+      );
+    }
+    const created = await db.Weet.create({ userId: user.id, content: data });
```

Comment:
Services should thought as final branches of a controller logic, so the logic that they execute should be very specific. As I mentioned to you before, the controller function should be the one that controls the flow of logic. This should be in a separate service function and should be called from the controller